src/event_pusher/mod_event_pusher_push.erl

```diff
@@ -273,18 +257,10 @@ parse_form(Form) ->
             invalid_form
     end.
 
--spec maybe_enable_node(jid:jid(), jid:jid(), pubsub_node(), form(), jlib:iq()) -> jlib:iq().
-maybe_enable_node(#jid{lserver = Host} = From, BarePubSubJID, Node, FormFields, IQ) ->
-    AllKnownDomains = ejabberd_router:dirty_get_all_domains() ++ virtual_pubsub_hosts(Host),
-    case lists:member(BarePubSubJID#jid.lserver, AllKnownDomains) of
-        true ->
-            ok = mod_event_pusher_push_backend:enable(jid:to_bare(From), BarePubSubJID, Node, FormFields),
-            store_session_info(From, {BarePubSubJID, Node, FormFields}),
-            IQ#iq{type = result, sub_el = []};
-        false ->
-            NewSubEl = [IQ#iq.sub_el, mongoose_xmpp_errors:remote_server_not_found()],
-            IQ#iq{type = error, sub_el = NewSubEl}
-    end.
+-spec enable_node(jid:jid(), jid:jid(), pubsub_node(), form(), jlib:iq()) ->
+    ok | {error, Reason :: term()}.
+enable_node(From, BarePubSubJID, Node, FormFields, IQ) ->
+    mod_event_pusher_push_backend:enable(jid:to_bare(From), BarePubSubJID, Node, FormFields).
```

Comment:
Why did we have this check in the code? I am not sure if it can be just removed, this changes the functionality.
Why not check it with `mongoose_domain_api:get_host_type` (and maybe use the version for subdomains that we are going to introduce)?

Reply:
The check is incorrect, it's against XMPP push XEP(s). we can specify the node from another federation here, and that check will fail. I think it was accidentally ported from some private code into open-source MIM.

Reply:
I don't know much about this functionality, by "against XMPP push XEP(s)" do you mean that XEP-0357 does not mention any checks on the server side in the [Enabling](https://xmpp.org/extensions/xep-0357.html#enabling) section?

Just asking: "node from another federation" - do you mean that the domain is not in the MIM routing table (yet)?

Reply:
s2s connections are not reflected in ejabberd_router tables at all, they are managed by `ejabberd_s2s` module. but even if don't have any established s2s connection yet, it doesn't mean that we should reject such a pubsub node. because normally s2s connection is established on the first routing.